src/main/java/frc/robot/utils/CRTGearRatioUtil.java

```diff
@@ -0,0 +1,154 @@
+    double sum = 0;
+    double a = secondaryGearRotationRatioTermSimplfied;
+    double b = mainGearRotationRatioTermSimplified;
+    for (int i = 1; i < b; i++) {
+      sum += ((a * (b - i) - 1.0) / b)
+          * (1 + Math.floor(((a * (b - i) - 1.0) / b)) - Math.ceil(((a * (b - i) - 1.0) / b)));
+    }
+    return (int) ((int) ((sum * b) + 1.0) / a); //basically, when you apply CRT, you can multiply by a certain factor (what this function returns, a constant for a given gear ratio) such that when using modding that factor times the output of crt and the simplified main gear rotation ratio term, it will correctly output a number between 0 and maxrotations. then, to account for wrap around and to make 0 full rotations work correctly, mod the main gear ratio term by the output + the simplified term
+  }
+
+  /**
+   * 
+   * @param mainGearRotationValue      Rotation value returned by the main gear
+   *                                   encoder
+   * @param secondaryGearRotationValue Rotation value returned by the secondary
+   *                                   gear encoder
+   * @return Returns the number of full rotations of the main gear
+   */
+  public int getMainGearFullRotations(double mainGearRotationValue, double secondaryGearRotationValue) {
+    double k = kFactor();
+    return (int) Math.round(mod(mainGearRotationRatioTerm, //mod the main gear rotation ratio term by the whole thing plus the simplified term to make 0 full rotations work properly
+        mod(k * //this is the multiplier, see the comment in the returns statement of kFactor() for more details
+        (mainGearRotationValue * secondaryGearRotationRatioTermSimplfied
+            - secondaryGearRotationValue * mainGearRotationRatioTermSimplified) //this returns a decimal where the multiplier is different for every gear ratio
+            , mainGearRotationRatioTermSimplified) //mod it with the main gear rotation ratio term so it is within the correct range
+            + mainGearRotationRatioTermSimplified)); //for more information, see the comment in the kFactor() function
+  }
+
+  /**
+   * 
+   * @return Returns true if the gear ratio works with the neccessary number of
+   *         max rotations, false otherwise
+   */
+  public boolean validateGearRatioWithMaxRotations() {
+    boolean works = true;
+    for (int i = 0; i < maxRotations * 6; i++) {
+      double elevatorPos = i * mainGearCircumference / 6; //set elevator positon (expected position that it will check with) through sixths of a rotation until max rotations is reached
+      double epoc = CRTGearRatioUtil.mod(elevatorPos / mainGearCircumference, 1) > 0.9 ? 0.0
+          : CRTGearRatioUtil.mod(elevatorPos / mainGearCircumference, 1) < 0.1 ? 0.0
+              : CRTGearRatioUtil.mod(elevatorPos / mainGearCircumference, 1); //calculate what the main gear encoder value would be at each position - the ternary operators account for annoying float division rounding problems on either end (0 and 1)
```

Comment:
How did you pick 0.9 and 0.1. Can you split these out into a variable reasonably? Seems like they only work because you picked the number 6, and increments of 1/6 go 0 to 0.16 (larger than 0.1).

Reply:
It seems like you actually just want to do "is approx 0" or "is approx 1" and this isn't a great way to do that. Also this would be clearer if you had ` CRTGearRatioUtil.mod(elevatorPos / mainGearCircumference, 1)` once instead of three times.

Reply:
done